dubbo-common/src/main/java/org/apache/dubbo/common/BaseServiceMetadata.java

```diff
@@ -42,7 +42,7 @@ public static String buildServiceKey(String path, String group, String version)
         if (version != null && version.length() > 0) {
             buf.append(":").append(version);
         }
-        return buf.toString();
+        return buf.toString().intern();
```

Comment:
I am a bit confused. Does the memory already being allocated from `toString()`?
I think doing intern this way will only possibly save the memory if the String returned will be held by a constant reference, it won't save the temporary memory allocated.

Reply:
> I think doing intern this way will only possibly save the memory if the String returned will be held by a constant reference, it won't save the temporary memory allocated.

Yes, it is useless about temporarory memory allocation. But, it can save persistence memory allocation. That is the urls from directory can hold the same String object, and temporarory String object from `toString()` can be release by GC. URLs will persistence exist if the invoker is still alive, and it will keep the reference of `ServiceKey` as a cache.

Reply:
I thought you were improving the efficiency of the RPC call, say, RT, QPS.